sections/nix_shell.zsh

```diff
@@ -0,0 +1,34 @@
+#
+# Nix Shell
+#
+# Nix can be used to provide some kind of virtual environment through the nix-shell command.
+# Link: https://nixos.org/manual/nix/stable/command-ref/nix-shell.html
+
+# ------------------------------------------------------------------------------
+# Configuration
+# ------------------------------------------------------------------------------
+
+SPACESHIP_NIXSHELL_SHOW="${SPACESHIP_NIXSHELL_SHOW=true}"
+SPACESHIP_NIXSHELL_PREFIX="${SPACESHIP_NIXSHELL_PREFIX="$SPACESHIP_PROMPT_DEFAULT_PREFIX"}"
+SPACESHIP_NIXSHELL_SUFFIX="${SPACESHIP_NIXSHELL_SUFFIX="$SPACESHIP_PROMPT_DEFAULT_SUFFIX"}"
+SPACESHIP_NIXSHELL_SYMBOL="${SPACESHIP_NIXSHELL_SYMBOL="❄ "}"
+SPACESHIP_NIXSHELL_COLOR="${SPACESHIP_NIXSHELL_COLOR="yellow"}"
+
+# ------------------------------------------------------------------------------
+# Section
+# ------------------------------------------------------------------------------
+
+# Shows whether nix-shell environment is active
+spaceship_nixshell() {
+  [[ $SPACESHIP_NIXSHELL_SHOW == false ]] && return
+
+  [[ -z $IN_NIX_SHELL ]] && return
```

Comment:
It would also be great to retrieve the value of `name` env, so we can show the `pure` and `impure` shell and their names.

It should display the following:

- `pure (name)` when `$name == "name"` in a pure nix-shell
- `impure (name)`  when `$name == "name"` in an impure nix-shell
- `pure` when `$name == ""` in a pure nix-shell
- `impure` when `$name == ""` in an impure nix-shell

Reply:
OK I'll look at replacing the "nix-shell" text with your suggestion

Reply:
You can extract shell type from `IN_NIX_SHELL` env variable.